src/app/shared/load-polygon-chart.directive.ts

```diff
@@ -91,5 +95,69 @@ export class LoadPolygonChartDirective implements OnChanges {
     g.append('path')
       .attr('d', areaBar(this.data))
       .style('fill', '#d9721f');
+
+    const focus = g.append('g').style('display', 'none');
+
+    focus.append('line')
+      .attr('id', 'focusLineX')
+      .attr('class', 'focusLine');
+
+    focus.append('circle')
+      .attr('id', 'focusCircle')
+      .attr('r', 1.5)
+      .attr('class', 'circle focusCircle');
+
+    focus.append('text')
+      .attr('id', 'focusText')
+      .attr('font-size', '10')
+      .attr('x', 1)
+      .attr('y', 8);
+
+    // @ts-ignore TS2345
+    const bisectDate = bisector((d, x) => +moment.utc(d[this.xkey]).isBefore(x)).right;
+
+    const rect = g.append('rect')
+      .attr('class', 'overlay')
+      .attr('width', this.width)
+      .attr('height', this.height)
+      .on('mouseover', () => focus.style('display', null))
+      .on('mouseout', () => focus.style('display', 'none'));
+
+    rect.on('mousemove', () => {
+      const coords = mouse(currentEvent.currentTarget);
+
+      const mouseDate: any = moment.utc(scaleX.invert(coords[0]));
+      // returns the index to the current data item
+      const i = Math.min(Math.max(bisectDate(this.appLoadPolygonChart, mouseDate), 0), this.appLoadPolygonChart.length - 1);
+      const d = this.appLoadPolygonChart[i];
+
+      const x = scaleX(isoParse(d[this.xkey]));
+      const y = scaleY(d[this.ykey] === undefined ? 0 : d[this.ykey]);
+
+      focus.select('#focusCircle')
+        .attr('cx', x)
+        .attr('cy', y);
+
+      focus.select('#focusLineX')
+        .attr('x1', x)
+        .attr('y1', scaleY(this.findMinY()))
+        .attr('x2', x)
+        .attr('y2', scaleY(this.findHighestY()));
+
+      const value = d[this.ykey] === undefined ? 0 : d[this.ykey];
+      const load = this.humanize.transform(value, this.measurement);
+
+      const dateToShow = this.dateFormat.transform(moment(d[this.xkey]).utc());
+      this.dataTooltip = d['NoData'] ? `No data at ${dateToShow}` : `${load} at ${dateToShow}`;
```

Comment:
Please check on existence and `null`.
If Point does not exists in DB it generated with ts, point number and time frame - no other fields at  this object.
Otherwise if point exit in DB, but no value for specific metrics - its field will be `null`.